src/components/leaderboards/Create.vue

```diff
@@ -0,0 +1,144 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+import { onBeforeRouteLeave, useRouter } from 'vue-router'
+import { useApi } from '../../composables/useApi'
+import { useAuth } from '../../composables/useAuth'
+import { useSessionToken } from '../../composables/useSessionToken'
+import { Leaderboards } from '../../lib/api/Leaderboards'
+import { CreateLeaderboardRequest } from '../../lib/api/data-contracts'
+
+const token = useSessionToken()
+const createError = ref('')
+const router = useRouter()
+
+const leaderboards = new Leaderboards({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const warnBeforeLeave = ref(true)
+
+const createRequest = ref<CreateLeaderboardRequest>({
+	name: '',
+	slug: '',
+	info: '',
+})
+
+onBeforeRouteLeave(() => {
+	if (
+		warnBeforeLeave.value &&
+		(createRequest.value.name ||
+		 createRequest.value.slug ||
+		 createRequest.value.info)
+	) {
+		if (!window.confirm('Do you want to leave? You have unsaved changes.')) {
+			return false
+		}
+	}
+})
+
+async function submit() {
```

Comment:
Since the await operator isn't used, this function doesn't need to be async. You can switch it to a regular function or await the call to `useApi()`.